### DISCUSSION TITLE
How to enable underline?

### DISCUSSION THREAD

**Discussion by SeanSelene** [wezterm/wezterm#6519] [Q&A]
### What Operating System(s) are you running on?

Windows

### WezTerm version

wezterm 20240203-110809-5046fc22

### Ask your question!

I learned wezterm has supported underline. It works on my Mac but not on Windows. I have tried  to run `wezterm -n` to skip my configuration. Here is the screenshot 
![image](https://github.com/user-attachments/assets/5d065141-43d1-4c4b-817a-19e00d31ea30)
Does it need some configuration to take effect?

**Comment by prosoitos** [comment]
I am also not sure how to get underlines in Wezterm. I don't seem to have any underline anywhere, even when I mouse hover hyperlinks.

OS: Arch Linux (6.13.2-arch1-1)
Wezterm: 20240203-110809-5046fc22

Thanks!!

**Comment by wez** [comment]
Note that conpty, a windows specific component, would swallow this sort of escape sequence before reporting it to wezterm. In the very latest nightly builds, conpty has been updated so you might find that this starts to work now.

**Reply by prosoitos** [reply]
And on Arch?

**Reply by prosoitos** [reply]
I am coming from alacritty and underlines were working

**Reply by prosoitos** [reply]
Note that this is not critical and a deal breaker. Mostly, I was wondering whether I had something to set in the config or whether it was supposed to work out of the box.

I now understand that it is the latter and that there must be something on my system preventing it.

I'd be curious to know how to fix it, but it might not be easy. Not sure whether other people on Arch Linux are facing the same problem.

**Reply by prosoitos** [reply]
I could open an issue about it if it were appropriate

**Reply by bew** [reply]
Yes please open a dedicated issue, Arch is a completely different environment, make sure to fill the issue template so we may reproduce the issue.

(Also, try to avoid making multiple comments just a few minutes apart, use the edit function instead.)

**Reply by prosoitos** [reply]
I found the problem: I am using Inconsolata as my font. Changing font solves the issue.

I suspect that the problem doesn't have anything to do with Wezterm then, but with the font itself and that I don't need to open an issue. Please let me know if I am wrong and should open the issue. Thank you.

In the meantime, because I really like Inconsolata, I will try to get my hyperlinks in a different color instead of being underlined (if I can figure it out...)

**Reply by DwcksDinh** [reply]
i have same problem with him my wezterm on windows doesn't show underline

**Reply by bew** [reply]
> i have same problem with him my wezterm on windows doesn't show underline

you gotta give more information about your environment if you want any help, like what font you're using..

**Reply by MiSawa** [reply]
FWIW, I had a similar problem with
```lua
  -- https://github.com/yuru7/bizin-gothic/releases/tag/v0.0.4
  font = wezterm.font('Bizin Gothic NF'),
  font_size = 13.0,
  treat_east_asian_ambiguous_width_as_wide = false,
```
, and adding
```lua
  underline_position = -2.0,
  underline_thickness = '1.2pt',
```
resolved the issue (the actual number would totally depend on your configuration). You might also want to tweak `line_height = 1.1`.

**Reply by prosoitos** [reply]
OMG! Thank you MiSawa!!! That solved the problem I had with Inconsolata! (moving the underline down is what resolved the problem with that font).
